Framework/PythonInterface/mantid/geometry/src/Exports/DetectorInfo.cpp

```diff
@@ -0,0 +1,71 @@
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidKernel/Quat.h"
+#include "MantidKernel/V3D.h"
+#include <boost/python/class.hpp>
+#include <boost/python/copy_const_reference.hpp>
+#include <boost/python/return_value_policy.hpp>
+
+using Mantid::Geometry::DetectorInfo;
+using Mantid::Kernel::Quat;
+using Mantid::Kernel::V3D;
+using namespace boost::python;
+
+void export_DetectorInfo() {
+
+  // Function pointers to distinguish between overloaded versions
+  bool (DetectorInfo::*isMonitor)(const size_t) const =
+      &DetectorInfo::isMonitor;
+
+  bool (DetectorInfo::*isMasked)(const size_t) const = &DetectorInfo::isMasked;
+
+  double (DetectorInfo::*twoTheta)(const size_t) const =
+      &DetectorInfo::twoTheta;
+
+  Mantid::Kernel::V3D (DetectorInfo::*position)(const size_t) const =
+      &DetectorInfo::position;
+
+  Mantid::Kernel::Quat (DetectorInfo::*rotation)(const size_t) const =
+      &DetectorInfo::rotation;
+
+  void (DetectorInfo::*setMasked)(const size_t, bool) =
+      &DetectorInfo::setMasked;
+
+  // Export to Python
+  class_<DetectorInfo, boost::noncopyable>("DetectorInfo", no_init)
+      .def("__len__", &DetectorInfo::size, (arg("self")),
+           "Returns the size of the DetectorInfo, i.e., the number of "
+           "detectors in the instrument.")
+
+      .def("size", &DetectorInfo::size, (arg("self")),
+           "Returns the size of the DetectorInfo, i.e., the number of "
+           "detectors in the instrument.")
+
+      .def("isMonitor", isMonitor, (arg("self"), arg("index")),
+           "Returns True if the detector is a monitor.")
+
+      .def("isMasked", isMasked, (arg("self"), arg("index")),
+           "Returns True if the detector is masked.")
+
+      .def("setMasked", setMasked, (arg("self"), arg("index"), arg("masked")),
+           "Set the mask flag of the detector where the detector is identified "
+           "by 'index'.")
+
+      .def("clearMaskFlags", &DetectorInfo::clearMaskFlags, (arg("self")),
+           "Sets all mask flags to false (unmasked).")
+
+      .def("isEquivalent", &DetectorInfo::isEquivalent,
+           (arg("self"), arg("other")), "Returns True if the content of this "
+                                        "detector is equivalent to the content "
+                                        "of the other detector.")
+
+      .def("twoTheta", twoTheta, (arg("self"), arg("index")),
+           "Returns 2 theta (scattering angle w.r.t beam direction).")
+
+      .def("position", position, (arg("self"), arg("index")),
+           "Returns the absolute position of the detector where the detector "
+           "is identified by 'index'.")
+
+      .def("rotation", rotation, (arg("self"), arg("index")),
+           "Returns the absolute rotation of the detector where the detector "
+           "is identified by 'index'.");
```

Comment:
Does it make sense to add methods that allow the user to iterate over a detector info? For example, instead of:

```python
for i in range(len(detinfo)):
   print (detinfo.position(i))
```
we do:
```python
for position in detinfo.positions():
  print (position)
```

Likewise what about getting a boolean numpy array of masked values?

Reply:
Yeah that sounds like a good idea - so you mean write a helper function for the `__iter__` method?

Reply:
Yea. I think boost should support returning an iterator. Maybe check with @OwenArnold first.

Reply:
I didn't think about that, but would seem to be a useful feature, especially on the python side. For good performance reason we have a design that is a SOA approach but a AOS approach is actually more natural for users.

Reply:
I can see that, but I can also see that users will start writing things like the above? I can see myself writing things like the above. And I don't think that would be the most efficient way to get all the positions.

Reply:
I have considered adding iterator support for `DetectorInfo`. I could be done similar to how @samueljackson92 implemented the iterator for `Histogram`. I would however recommend to do this as a separate pull request, if anyone feels the need.

Reply:
@martyngigg